src/Groups/matrices/linear_isconjugate.jl

```diff
@@ -0,0 +1,155 @@
+# TODO: in this file are used many methods TEMPORARILY defined in files matrix_manipulation.jl and stuff_field_gen.jl
+# once methods in those files will be deleted / replaced / modified, this file need to be modified too
+
+
+
+export
+    generalized_jordan_block,
+    generalized_jordan_form,
+    isconjugate,
+    issemisimple,
+    isunipotent,
+    pol_elementary_divisors,
+    multiplicative_jordan_decomposition
+
+
+
+
+########################################################################
+#
+# Semisimple / Unipotent
+#
+########################################################################
+
+"""
+    multiplicative_jordan_decomposition(M::MatrixGroupElem)
+Return `S` and `U` in the group `G = parent(M)` such that `S` is semisimple, `U` is unipotent and  `M = SU = US`.
+!!! warning "WARNING:" 
+    this is *NOT*, in general, the same output returned when `M` has type `MatElem`.
+"""
+function multiplicative_jordan_decomposition(x::MatrixGroupElem)
+   a = order(x)
+   p = characteristic(base_ring(x))
+   alpha = valuation(a,p)
+   m = div(a, p^alpha)
+   k = crt(fmpz(0),fmpz(p^alpha),fmpz(1),fmpz(m))
+#   a,b = multiplicative_jordan_decomposition(x.elm)
+   return x^k, x^(a+1-k)
+end
+
+"""
+    issemisimple(x::MatrixGroupElem)
+Return whether `x` is semisimple, i.e. has order coprime with the characteristic of its base ring.
+"""
+issemisimple(x::MatrixGroupElem) = iscoprime(Int(order(x)::fmpz), Int(characteristic(x.parent.ring)))
+
+"""
+    isunipotent(x::MatrixGroupElem)
+Return whether `x` is unipotent, i.e. its order is a power of the characteristic of its base ring.
+"""
+isunipotent(x::MatrixGroupElem) = isone(x) || ispower(Int(order(x)))[2]==Int(characteristic(x.parent.ring))
+
+
+
+
+
+
+########################################################################
+#
+# MyIsConjugate
+#
+########################################################################
+
+
+
+# return an element in the centralizer of x in GL(n,F) with determinant d
+
+# first: brute force way
+# x has type MatrixGroupElem
+function _elem_given_det(x,d)
```

Comment:
"brute force way" of what? does that refer to the function `_elem_given_det`? What does it do?!

Reply:
Yes. To find an element in the centralizer of `x`, it computes the generators of the centralizer and multiplies them in order to obtain an element with the appropriate determinant, that is not the cleanest way to proceed. Do I specify this in the comment?

Reply:
Yes, all this extra information would be useful in a comment

Reply:
Added it.